txscript/standard.go

```diff
@@ -350,6 +350,21 @@ func PayToAddrScript(addr btcutil.Address) ([]byte, error) {
 	return nil, ErrUnsupportedAddress
 }
 
+// NullDataScript creates a provably prunable script
+// containing OP_RETURN followed by the passed data.
+func NullDataScript(data []byte) ([]byte, error) {
+	if len(data) > MaxDataCarrierSize {
+		return nil, ErrStackLongScript
+	}
+
+	script := make([]byte, 3+len(data))
```

Comment:
In case he above wasn't clear, here is some code using both that should illustrate better:

```Go
func main() {
	data := hexToBytes("0102030405060708091a1b1c1d1e1f101112")
	script, _ := txscript.NewScriptBuilder().AddOp(txscript.OP_RETURN).AddData(data).Script()
	disasm, _ := txscript.DisasmString(script)
	fmt.Printf("%x\n", script)
	fmt.Println(disasm)

	script, _= NullDataScript(data)
	disasm, _ = txscript.DisasmString(script)
	fmt.Printf("%x\n", script)
	fmt.Println(disasm)}
```

Output:
```
6a120102030405060708091a1b1c1d1e1f101112
OP_RETURN 0102030405060708091a1b1c1d1e1f101112
6a4c120102030405060708091a1b1c1d1e1f101112
OP_RETURN 0102030405060708091a1b1c1d1e1f101112
```

Notice how they both disassemble to the same thing, but the script builder version is the shorter, canonical form.